src/main/java/org/apache/ibatis/migration/CommentStrippingReader.java

```diff
@@ -0,0 +1,39 @@
+/**
+ *    Copyright 2010-2018 the original author or authors.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package org.apache.ibatis.migration;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.Reader;
+
+/**
+ * @author cbongiorno on 8/16/18.
+ */ // This little trick is necessary as a maven plugin sticks headers on all of our files.
+class CommentStrippingReader extends BufferedReader {
```

Comment:
You're forgetting about multiline comments, and multiline comment _directives_.  The parser handles that already, though, so this class should be completely unnecessary.

Reply:
Actually, it's not. It's not meant to be an all-encompassing solution. It solves the problem where a maven plugin puts a big preamble on every file. It's mean to only deal with SQL ```--``` comments. Without this, it was a bitch to get your byte test working